DynamicMeter.c

```diff
@@ -97,15 +98,18 @@ static const char* DynamicMeter_getCaption(const Meter* this) {
 }
 
 static void DynamicMeter_getUiName(const Meter* this, char* name, size_t length) {
+   assert(length > 0);
+
    const Settings* settings = this->host->settings;
    const DynamicMeter* meter = Hashtable_get(settings->dynamicMeters, this->param);
    if (meter) {
       const char* uiName = meter->caption;
       if (uiName) {
-         int len = strlen(uiName);
+         size_t len = strlen(uiName);
          if (len > 2 && uiName[len - 2] == ':')
             len -= 2;
-         xSnprintf(name, length, "%.*s", len, uiName);
+
+         xSnprintf(name, MINIMUM(length, len + 1), "%s", uiName);
```

Comment:
Why this change?

Reply:
@BenBE The "precision" part of the format specifier takes type `int` only and not `size_t`. Since I am processing string length now in `size_t`, changing the length argument to this avoids an `int` downcast and a `len <= INT_MAX` assumption.

Reply:
Couldn't this be written as a `xStrncpy` in that case instead?

Reply:
@BenBE You mean `String_safeStrncpy`, but good call anyway.